lib/SPIRV/SPIRVReader.cpp

```diff
@@ -4318,6 +4318,50 @@ bool SPIRVToLLVM::transMetadata() {
       F->setMetadata(kSPIR2MD::IntelFPGAIPInterface,
                      MDNode::get(*Context, InterfaceMDVec));
     }
+    if (auto *EM = BF->getExecutionMode(
```

Comment:
Should we check to ensure that only one mode is chosen?

We should use "else if" for the second and third cases.

Consider refactoring common code.

Reply:
Actually no. Even so it seems to have no sense to enable several different `maxregisters` execution modes, neither SPIR-V core spec, nor extension prohibits usage of several execution modes at the same time. So, I believe this code should not be changed to `else if`.

Reply:
From https://github.com/KhronosGroup/SPIRV-Registry/pull/235, in extensions/INTEL/SPV_INTEL_maximum_registers.asciidoc

* Each *OpEntryPoint* must contain at most one of the
*MaximumRegistersINTEL*, *MaximumRegistersIdINTEL*, or
*NamedMaximumRegistersINTEL* execution modes.

Reply:
Applied in 84a3ead. Unfortunately I can't add meaningful tests for that, because llvm-spirv fails with an assertion in case we add more than one execution mode of the certain type. 
Verified locally that this check both on forward and reverse translation in case of invalid IR/SPV.